src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs

```diff
@@ -9781,5 +9782,81 @@ public async Task MulticastDelegateConstraint()
 ";
             await VerifyItemExistsAsync(markup, "MulticastDelegate");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.TargetTypedCompletion)]
+        public async Task TestTargetTypeFilterWithExperimentEnabled()
+        {
+            var markup =
+@"public class C
+{
+    int intField;
+    void M(int x)
+    {
+        M($$);
+    }
+}";
+            await VerifyItemExistsAsync(
+                markup, "intField",
+                matchingFilters: new List<CompletionItemFilter> { CompletionItemFilter.FieldFilter, CompletionItemFilter.TargetTypedFilter },
+                targetTypedFilterExperimentEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.TargetTypedCompletion)]
+        public async Task TestNoTargetTypeFilterWithExperimentDisabled()
+        {
+            var markup =
+@"public class C
+{
+    int intField;
+    void M(int x)
+    {
+        M($$);
+    }
+}";
+            await VerifyItemExistsAsync(
+                markup, "intField",
+                matchingFilters: new List<CompletionItemFilter> { CompletionItemFilter.FieldFilter },
+                targetTypedFilterExperimentEnabled: false);
+        }
+
+
+        [Fact, Trait(Traits.Feature, Traits.Features.TargetTypedCompletion)]
+        public async Task TestTargetTypeFilter_NotOnObjectMembers()
+        {
+            var markup =
+@"public class C
+{
+    void M(int x)
+    {
+        M($$);
+    }
+}";
+            await VerifyItemExistsAsync(
+                markup, "GetHashCode",
+                matchingFilters: new List<CompletionItemFilter> { CompletionItemFilter.MethodFilter },
+                targetTypedFilterExperimentEnabled: true);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.TargetTypedCompletion)]
+        public async Task TestTargetTypeFilter_NotNamedTypes()
+        {
+            var markup =
+@"public class C
+{
+    void M(C c)
+    {
+        M($$);
+    }
+}";
+            await VerifyItemExistsAsync(
+                markup, "c",
+                matchingFilters: new List<CompletionItemFilter> { CompletionItemFilter.LocalAndParameterFilter, CompletionItemFilter.TargetTypedFilter },
+                targetTypedFilterExperimentEnabled: true);
+
+            await VerifyItemExistsAsync(
+                markup, "C",
+                matchingFilters: new List<CompletionItemFilter> { CompletionItemFilter.ClassFilter },
+                targetTypedFilterExperimentEnabled: true);
+        }
```

Comment:
seems liek a very small amount of tests.  do we want to have at least a few more scenarios?

Reply:
My initial thought was that I wanted to test here with a couple basic scenarios to make sure this feature works at all. Given I'm strictly using the position-based TypeInferrer right now, any problems would be attributed to the TypeInferrer, or would require a change in my approach here, so I would add the tests to the appropriate spot at that point. 

And we have lots of TypeInferrer tests. I guess the question is if they're testing the scenarios we care about here. Anyway, I'm happy to grow this test list, I'm just not sure where to draw the line to avoid duplicating TypeInferrer tests, and to avoid having the tests in the wrong location. Thoughts @CyrusNajmabadi ?